### ISSUE TITLE
vue/html-indent is invalid in the node api

### ISSUE THREAD

**Issue by dreamerldq** [vuejs/eslint-plugin-vue#2084]
**Checklist**

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version: ^8.33.** 
- **eslint-plugin-vue version: ^9.9.0** 
- **Node version: v14.17.3** 
- **Operating System: MacOS 13.2 (22D49)** 

**Please show your full configuration:**

```json5
    useEslintrc: false,
        fix: true,
        overrideConfig: {
            parser: 'vue-eslint-parser',
            extends: ['plugin:vue/recommended', 'eslint:recommended'],
            rules: {
                'vue/comment-directive': 'off',
                'vue/block-tag-newline': 'error',
                'vue/html-self-closing': 'error',
                'vue/html-indent': ["error", 2, {
                    'attribute': 1,
                    'baseIndent': 1,
                    'alignAttributesVertically': false
                }],
                'indent': ['error', 4, {SwitchCase: 1}],
                'quotes': ['error', 'single', {avoidEscape: false, allowTemplateLiterals: true}],
                'quote-props': ['error', 'consistent-as-needed', { 'keywords': true }],

            },
        },
```

**What did you do?**	

I'm using the eslint node api to do autofix
```vue
const { ESLint } = require('eslint');
const fs  = require('fs');
const testCode =
`<template><div class="app">
    <XXRank
    :rank-type="rankType"
    :ranks="ranks"
    :desc="desc"
    :value-icon="valueIcon"
    :show-vip-icon="false"
    :list-config="listConfig"
    @refresh="handleRefresh"
  >
      <template #rank-time>
      
      
      
      
      
      
      
        <RankTime
    v-if="RankTimeConfig.countTime"
    :unit-color="RankTimeConfig.unitColor"
    :num-color="RankTimeConfig.numColor"
    :count-time="RankTimeConfig.countTime"
    :time-type="RankTimeConfig.timeType"
    :num-bg="RankTimeConfig.numBg"
    :label-image="RankTimeConfig.labelImage"
    @refresh="handleRefresh"
  />
      </template>

    </XXRank>
  </div>
</template>
`;

(async function main() {
    // 1. Create an instance
    const eslint = new ESLint({
        useEslintrc: false,
        fix: true,
        overrideConfig: {
            parser: 'vue-eslint-parser',
            extends: ['plugin:vue/recommended', 'eslint:recommended'],
            rules: {
                'vue/comment-directive': 'off',
                'vue/block-tag-newline': 'error',
                'vue/html-self-closing': 'error',
                'vue/html-indent': ["error", 2, {
                    'attribute': 1,
                    'baseIndent': 1,
                    'alignAttributesVertically': false
                }],
                'indent': ['error', 4, {SwitchCase: 1}],
                'quotes': ['error', 'single', {avoidEscape: false, allowTemplateLiterals: true}],
                'quote-props': ['error', 'consistent-as-needed', { 'keywords': true }],

            },
        },
    });

    // 2. Lint text.
    const results = await eslint.lintText(testCode);
    console.log(results[0].output);
    fs.writeFileSync('index.vue', results[0].output)


})().catch((error) => {
    process.exitCode = 1;
    console.error(error);
});

```

**What did you expect to happen?**

I expect to be able to format the template properly
```vue
<template>
  <div class="app">
    <XXRank
      :rank-type="rankType"
      :ranks="ranks"
      :desc="desc"
      :value-icon="valueIcon"
      :show-vip-icon="false"
      :list-config="listConfig"
      @refresh="handleRefresh"
    >
      <template #rank-time>
        <RankTime
          v-if="RankTimeConfig.countTime"
          :unit-color="RankTimeConfig.unitColor"
          :num-color="RankTimeConfig.numColor"
          :count-time="RankTimeConfig.countTime"
          :time-type="RankTimeConfig.timeType"
          :num-bg="RankTimeConfig.numBg"
          :label-image="RankTimeConfig.labelImage"
          @refresh="handleRefresh"
        />
      </template>
    </XXRank>
  </div>
</template>

```

**What actually happened?**
```vue
<template>
<div class="app">
    <XXRank
    :rank-type="rankType"
    :ranks="ranks"
    :desc="desc"
    :value-icon="valueIcon"
    :show-vip-icon="false"
    :list-config="listConfig"
    @refresh="handleRefresh"
  >
      <template #rank-time>
<RankTime
    v-if="RankTimeConfig.countTime"
    :unit-color="RankTimeConfig.unitColor"
    :num-color="RankTimeConfig.numColor"
    :count-time="RankTimeConfig.countTime"
    :time-type="RankTimeConfig.timeType"
    :num-bg="RankTimeConfig.numBg"
    :label-image="RankTimeConfig.labelImage"
    @refresh="handleRefresh"
  />
      </template>
</XXRank>
  </div>
</template>


```

**Repository to reproduce this issue**
[playground](https://ota-meshi.github.io/eslint-plugin-vue-demo/#eJyNkj1vgzAQhv8KcmeXPaJV1UxZowwZWBxzgFVjLGyIqij/vXd2DKRNpbBw9z6vfB/2hcm+ArZhhYfOauHhvTRFpaZMauHcW8mEtSVDMcOvOB73wnzFZDNgyP23BXRRfMCwZCvobsDNagVOoki/WZuEHoEr2RskIdlhPGPX9mc+KZsctdBuKaOV8xxBrRpklG1DkgwfA9QDuBZhK0ylYR/zyG9j4WBp+uwlTqW6sAia9oBx9E1c1XhQEmOlV9mPxgdTamo0iprS/fDXTWxLaHabsfvPjOjeG2qF7p7pg3zpgn6ZCd3dF3Vxah628Nks6xYn0Fx1onlwZoA7Ys9tP1/Wn69fXx5fWaBFjm8xiCsHu/4Ao+ro+g==)

**Comment by ota-meshi** [comment]
The repro demo you provided did not enable `vue/html-indent` rule.
If I enable `vue/html-indent` rule it seems to work fine.

https://ota-meshi.github.io/eslint-plugin-vue-demo/#eJyNkk1vgzAMhv9KlB0qVbBoOyJWbeup16qaKo0dAhiIFgJKAlNV9b8vHwulWyeVC7afN47t+IiLrgSc4FQVkvV6lQmyRKA4ExotxgFIo1seM1GC0IsEvWcYpOxkhiP0GKFjJpD7FlRryfJBgxE9RFM4pwo24fAsTjmrxUs4o95AalZQzg9GVlGuwCpPH2hJMpGSqbZUQ9tzqsHaJRtRwalSTxmmfZ/hlU+f7vdbKj69k0hjxvrQg1FZe2fMDM+g+gFqipagChO0vyk2Uj5AzIpOGOKcjbEnrJruKx5ZHxS+h0A5Uzo2oGK1YdZbOycIniVUElRjYENFyWHrfc9XYWpT9+jOd8VaNwjb7c7YXjfGrDKJQtDfdF90g9BOFIoaBLNF8U7+VVu2tmhSi6H9T2zQpdbd5aq7pQ6rCw/0S2zRxXvZKvL6agmv9XncNAezsy2tr+R0cGPZbdMn5/GT+fYRv2WOpsTsogvOFDjCcuCgcHI8nb4B+Msh1A==

**Comment by dreamerldq** [comment]
Thanks for your reply! Yes, starting rule in the repo demo can solve the problem, in fact, the same configuration in vscode can also be solved, but the core of my problem lies in the use of eslint node api, did not achieve the expected results, you can look at the "what did you do" in The code I attached, after running, does not format correctly into the code in "what did you expect to happen"

**Comment by dreamerldq** [comment]
> 

repro demo: https://github.com/dreamerldq/eslint-problem

**Comment by ota-meshi** [comment]
I think you need to give `options.filePath`.

https://eslint.org/docs/latest/integrate/nodejs-api#-eslintlinttextcode-options
